packages/mcp/src/utils/auth.ts

```diff
@@ -152,3 +152,60 @@ export async function getDefaultTargetOrg(): Promise<OrgConfigInfo | undefined>
 export async function getDefaultTargetDevHub(): Promise<OrgConfigInfo | undefined> {
   return getDefaultConfig(OrgConfigProperties.TARGET_DEV_HUB);
 }
+
+/**
+ * Validates that all allowed orgs are sandboxes (not production orgs)
+ * by querying the Organization.IsSandbox field via SOQL.
+ *
+ * @param allowedOrgs - Array of allowed org authorizations to validate
+ * @throws {Error} If any org is found to be a production org (IsSandbox = false)
+ */
+export async function validateSandboxOrgs(allowedOrgs: SanitizedOrgAuthorization[]): Promise<void> {
+  const productionOrgs: string[] = [];
+
+  // eslint-disable-next-line no-await-in-loop
+  for (const org of allowedOrgs) {
```

Comment:
We could possibly speed up this section/logic by putting this unit of work into it's own promise and doing `Promise.all()`. Might get rid of the need for all these eslint disables!